Sources/Datadog/Tracer.swift

```diff
@@ -226,7 +220,7 @@ public class Tracer: OTTracer {
             tracer: self,
             context: spanContext,
             operationName: operationName,
-            startTime: startTime ?? dateProvider.currentDate(),
+            startTime: startTime ?? core.v1.context?.dateProvider.currentDate() ?? Date(),
```

Comment:
We should be able to easily get non-optional context at this point. The `core.v1.context` should be captured at `Tracer.init`. If it resolves to `nil` it means that the core is not set, so the user makes a mistake of creating `Tracer` without initialising the SDK first (should be alerted with printing configuration error). Otherwise, the context will resolve to a value and we will be able to get rid of this implicit unwrapping here.